src/oge/output_data.py

```diff
@@ -636,3 +612,18 @@ def add_generated_emission_rate_columns(df):
                 path_prefix,
                 skip_outputs,
             )
+
+
+def add_generated_emission_rate_columns(df):
```

Comment:
Missing docstring. 

Also it looks like this removes the original behavior to fill missing emission values with zero. Why was that changed?
```
# where the rate is missing because of a divide by zero (i.e.
                        # net_generation_mwh is zero), replace the emission rate with
                        # zero. We want to keep all other NAs so that they get flagged
                        # by our validation checks since this indicates an unexpected
                        # issue
                        df.loc[df["net_generation_mwh"] == 0, col_name] = df.loc[
                            df["net_generation_mwh"] == 0, col_name
                        ].fillna(0)
```

Reply:
It comes from the first commit: https://github.com/singularity-energy/open-grid-emissions/pull/295/commits/dd1b9203284a42b4ef70c9abfbcf3540ffceb97e

I did not make this edit.

Reply:
It looks like you factored out the `add_generated_emission_rate_columns` function from the ` write_power_sector_results` function and you refactored it by adding the `fillna(0)` right before the 2 `replace`. I believe it is equivalent than adding the extra assignment:
```
df.loc[df["net_generation_mwh"] == 0, col_name] = df.loc[
    df["net_generation_mwh"] == 0, col_name
].fillna(0)
```

Reply:
Ok maybe we leave the update then. I guess this would be consistent with having a missing emissions rate if there is no generation data to report, rather than a zero rate, which could be confused with zero-carbon

Reply:
Changed to what we have in `development`/`main`. As a result, the warnings that were raised for some BA, such as:
```
2024-05-14 12:23:37 [INFO] oge.oge.validation:222 Checking that fuel and emissions values are positive...  
2024-05-14 12:23:37 [INFO] oge.oge.validation:296 OK
2024-05-14 12:23:37 [INFO] oge.oge.validation:302 Checking that no values are missing...  
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_co2_rate_lb_per_mwh_for_electricity is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_co2e_rate_lb_per_mwh_for_electricity is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_nox_rate_lb_per_mwh_for_electricity is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_so2_rate_lb_per_mwh_for_electricity is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_co2_rate_lb_per_mwh_for_electricity_adjusted is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_co2e_rate_lb_per_mwh_for_electricity_adjusted is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_nox_rate_lb_per_mwh_for_electricity_adjusted is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:308 There are 4 records where generated_so2_rate_lb_per_mwh_for_electricity_adjusted is missing.
2024-05-14 12:23:37 [WARNING] oge.oge.validation:318 The above missing values are errors and must be fixed
```
now disappeared